### PR TITLE
SorrTask 420 Create empty Google sheet

### PR DESCRIPTION
**Pull request by yiyunlei**
#420
Since I can't find a library that includes all the functions I need, I've decided to write a Google API library. I believe we can start the first round of review, even though I still need to modify the token file paths. These pieces of code are very close to the final version.

### CODE REVIEW

File: linkedin/google_api/google_sheets.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
```
Comment (gpsaggese) [inline linkedin/google_api/google_sheets.py:1]: I know it's a draft but I am a curious person by nature.

File: linkedin/google_api/google_sheets.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# Scopes required to make both API calls -> https://developers.google.com/docs/api/reference/rest/v1/documents/create
+SCOPES = ["https://www.googleapis.com/auth/drive"]
+
+
+# #############################################################################
+# TODO(Yiyun): Consider if add function of getting folder id from shared dir name. (Folder name and file name are not unique.)
+class googleSheetAPI:
```
Comment (gpsaggese) [inline linkedin/google_api/google_sheets.py:16]: -> GoogleSheetApi (according to our naming scheme)

File: linkedin/google_api/google_sheets.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    def create_empty_google_file(self, gsheets_name: str, user: str, folder_id: str) -> None:
+        """
+        Create an empty google file and share it to a user
```
Comment (gpsaggese) [inline linkedin/google_api/google_sheets.py:19]: In comments google -> Google

File: linkedin/google_api/google_sheets.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                    'title': gsheets_name
+                }
+            }
+            spreadsheet = gspread_service.spreadsheets().creat(
+                body=spreadsheet,
+                fields='spreadsheetId'
+            ).execute()
+            # Create a new document with a determined name. The contents of the document can be set using additional API calls.
+            spreadsheet_id = spreadsheet.get('spreadsheetId')
+            # print('The file id of the new google sheet is: {}'.format(spreadsheet_id))
```
Comment (gpsaggese) [inline linkedin/google_api/google_sheets.py:53]: You can use _LOG.debug which makes it simpler to debug

Even if this is a library you can use some of the infra from
./dev_scripts/script_skeleton.py
to make it easier to run (e.g., you can have a lib and then a script that calls into that if you want to test)

**Review by gpsaggese — COMMENTED** [review]
Looking good. Happy to see that you know how to write nice code.

File: linkedin/google_api/google_sheets.py (hunk 1 above)
Comment (yiyunlei) [reply linkedin/google_api/google_sheets.py:1]: 🤣

File: linkedin/google_api/google_sheets.py (hunk 4 above)
Comment (yiyunlei) [reply linkedin/google_api/google_sheets.py:53]: > You can use _LOG.debug which makes it simpler to debug

Yes, I forgot to mention that step. I'm going to update the code to follow the Sorrentum code style. 

> Even if this is a library you can use some of the infra from ./dev_scripts/script_skeleton.py to make it easier to run (e.g., you can have a lib and then a script that calls into that if you want to test)

Good suggestion! I will add a test script.

Comment (yiyunlei) [reply linkedin/google_api/google_sheets.py]: I've decided to use notebooks to run it, as they're more convenient.

File: linkedin/google_api/google_file_api.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# Scopes required making API calls.
+_LOG = logging.getLogger(__name__)
+SCOPES = ["https://www.googleapis.com/auth/drive"]
```
Comment (gpsaggese) [inline linkedin/google_api/google_file_api.py:14]: If it's used only in one function let's move it there (together with the comment)

File: linkedin/google_api/google_file_api.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        """
+        Create a new Google sheet.
+        """
+        return self._create_new_google_document(gsheet_name, "sheets")
+
+    def _create_new_google_doc(self, gdoc_name: str) -> str:
+        """
+        Create a new Google doc.
+        """
+        return self._create_new_google_document(gdoc_name, "docs")
```
Comment (gpsaggese) [inline linkedin/google_api/google_file_api.py:184]: Little thin. I would inline in the caller

File: linkedin/google_api/google_file_api.py [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+
+    def _share_google_file(self, gsheet_id: str, user: str) -> None:
+        # Create the permission.
```
Comment (gpsaggese) [inline linkedin/google_api/google_file_api.py:187]: let's add a docstring

File: linkedin/google_api/google_file_api.py [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+            .create(fileId=gsheet_id, body=parameters)
+            .execute()
+        )
+        _LOG.info(
+            "The new permission id of the document is: '%s'",
+            new_permission.get("id"),
+        )
+        _LOG.info("The google file is shared to '%s'.", user)
+
+    def _move_gfile_to_dir(self, gfile_id: str, folder_id: str) -> dict:
```
Comment (gpsaggese) [inline linkedin/google_api/google_file_api.py:200]: is this gsheet_id? Let's try to call the same things with the same names

File: linkedin/google_api/google_file_api.py [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+        """
+        Moves a Google file to a specified folder in Google Drive.
```
Comment (gpsaggese) [inline linkedin/google_api/google_file_api.py:202]: -> Move

and some comments on the params

File: linkedin/google_api/google_file_api.py [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+                addParents=folder_id,
+                removeParents="root",
+                supportsAllDrives=True,
+            )
+            .execute()
+        )
+        return res
+
+    def _get_folders_in_gdrive(self) -> list:
+        response = (
```
Comment (gpsaggese) [inline linkedin/google_api/google_file_api.py:218]: let's add docstring with params and also examples of how the variables look like

**Review by gpsaggese — COMMENTED** [review]
Good work. Let's merge this as it is and then you can do a follow up PR with the suggested changes

**Pull request merged**